packages/shorebird_tests/test/ios_test.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:test/test.dart';
+
+import 'shorebird_tests.dart';
+
+void main() {
+  group('shorebird ios projects', () {
+    testWithShorebirdProject('can build', (projectDirectory) async {
+      await projectDirectory.runFlutterBuildIos();
+
+      expect(projectDirectory.iosArchiveFile().existsSync(), isTrue);
+      expect(projectDirectory.getGeneratedIoShorebirdYaml(), completes);
+    });
+
+    group('when passing the public key through the environment variable', () {
+      testWithShorebirdProject(
+        'adds the public key on top of the original file',
+        (projectDirectory) async {
+          final originalYaml = projectDirectory.shorebirdYaml;
+
+          const base64PublicKey = 'public_123';
+          await projectDirectory.runFlutterBuildIos(
+            environment: {
+              'SHOREBIRD_PUBLIC_KEY': base64PublicKey,
+            },
+          );
+
+          final generatedYaml =
+              await projectDirectory.getGeneratedIoShorebirdYaml();
+
+          expect(
+            generatedYaml.keys,
+            containsAll(originalYaml.keys),
+          );
+
+          print(generatedYaml);
+          expect(
+            generatedYaml['patch_public_key'],
+            equals(base64PublicKey),
+          );
+        },
+      );
+    });
+
+    // TODO(erickzanardo): Add tests for flavors.
```

Comment:
Dynamically adding flavors to an iOS doesn't seems to be simple, so to not put too much time in it, for now I just tested the cases that there are no flavors. Happy to revisit this though

Reply:
Yeah, this is normally set up through xcode. I'm sure there's some `xcrun` command we could run or scheme file we could copy, but agreed that it's probably not within the scope of this change.

Reply:
yep. I tried looking for a xrun command but I couldn't find any 🙃 

There is an interesting flutter package called `flutter_flavorizr` which I might give it a try later. I just felt like not putting much more time in it so we can get to the demo soon!

But I plan on coming back here, since I feel these are important tests to make!